cogs/member_stats.py

```diff
@@ -1,15 +1,141 @@
 import discord
-from discord.ext import commands
+from discord.ext import tasks, commands
 from pathlib import Path
-from .validation import is_moderator, is_mod_commands_channel
+from .validation import (is_moderator, is_mod_commands_channel,
+                         asyncless_is_moderator,
+                         asyncless_is_mod_commands_channel,
+                         is_process_and_user_clear)
 import json
+import os
+from os import path
+from datetime import datetime
+from dateutil import tz
 
 
 class MemberStats(commands.Cog):
+
     def __init__(self, bot):
         self.bot = bot
+        self.isRecording = False
+
+    def cog_unload(self):
+        self.m_stats.cancel()
+
+    async def is_wanting_cancel(self, message, command):
+        if message.content == 'cancel':
+            await message.channel.send('Cancelling the procecss.')
+            self.bot.processes[command] = None
+            return True
+        return False
+
+    @commands.command()
+    async def m_stats_start(self, ctx):
+        if not await is_moderator(ctx):
+            return
+        if not await is_mod_commands_channel(ctx):
+            return
+
+        author_id = ctx.author.id
+        if not await is_process_and_user_clear(self.bot, 'm_stats_start', ctx):
+            return
```

Comment:
Generally, we're okay for users to engage with Seng in non-conversation commands, whenever they like, including if that user is in a conversation (process). This enables users to retrieve (for instance) some information from Seng, but without interrupting a conversation they're also having with Seng. 

However, the rule is that a user cannot be engaged with more than one conversation with Seng, at any one time.

As this method has no conversation, then there's no need to check if the user is already in a process or not.

Maybe we'll change this design in the future though.

(Note that this comment also applies to some of the other `is_process_and_user_clear` calls)